packages/grafana-data/src/transformations/transformers/calculateField.ts

```diff
@@ -249,6 +258,33 @@ export const calculateFieldTransformer: DataTransformerInfo<CalculateFieldTransf
                 fields: options.replaceFields ? [f] : [...frame.fields, f],
               };
             });
+          case CalculateFieldMode.TemplateExpression:
+            if (options.template?.expression !== undefined) {
+              return data.map((frame) => {
+                const newFieldVals = Array.from({ length: frame.length }, (_, i) => {
+                  const fieldVars: ScopedVars = {};
+                  frame.fields.forEach((field) => {
+                    fieldVars[field.name] = {
+                      value: field.values[i],
+                    };
+                  });
+                  const replaced = ctx.interpolate(options.template!.expression, fieldVars);
```

Comment:
This is where the added variables list passed through to the interpolate function is used. We iterate through the frames, and the fields in the frames, and add the field name/value pair to the variables list to be interpolated.